aztec/src/main/kotlin/org/wordpress/aztec/formatting/InlineFormatter.kt

```diff
@@ -250,10 +247,33 @@ class InlineFormatter(editor: AztecText, val codeStyle: CodeStyle, private val h
         }
     }
 
-    private fun applyMarkInlineStyle(start: Int = selectionStart, end: Int = selectionEnd) {
-        val previousSpans = editableText.getSpans(start, end, MarkSpan::class.java)
-        previousSpans.forEach {
-            it.applyInlineStyleAttributes(editableText, start, end)
+    private fun applyAfterMarkInlineStyle(start: Int = selectionStart, end: Int = selectionEnd) {
+        // If there's no new mark style color to update, it skips applying the style updates.
+        if (markStyleColor == null) {
+            return
+        }
+
+        val spans = editableText.getSpans(start, end, MarkSpan::class.java)
+        spans.forEach { span ->
+            if (span != null) {
+                val color = span.getTextColor()
+                val currentSpanStart = editableText.getSpanStart(span)
+
+                editableText.removeSpan(span)
+                editableText.setSpan(
+                    MarkSpan(AztecAttributes(), color),
+                    currentSpanStart,
+                    start,
+                    Spanned.SPAN_EXCLUSIVE_EXCLUSIVE
+                )
```

Comment:
When there's an existing Mark style but the color changes:

This will keep the previous Mark style, from where it starts `currentSpanStart` until the new Mark style starts `start`.